packages/vm/src/calls.rs

```diff
@@ -441,6 +461,47 @@ where
     )
 }
 
+/// Calls Wasm export "migrate" and returns raw data from the contract.
+/// The result is length limited to prevent abuse but otherwise unchecked.
+/// The difference between this function and [call_migrate_raw] is the
+/// additional argument - `migrate_info`. It contains additional data
+/// related to the contract's migration procedure.
+///
+/// It is safe to call this method instead of [call_migrate_raw] even
+/// if a contract contains the migrate entrypoint without `migrate_info`.
+/// In such case this structure is omitted.
+pub fn call_migrate_with_info_raw<A, S, Q>(
+    instance: &mut Instance<A, S, Q>,
+    env: &[u8],
+    msg: &[u8],
+    migrate_info: &[u8],
+) -> VmResult<Vec<u8>>
+where
+    A: BackendApi + 'static,
+    S: Storage + 'static,
+    Q: Querier + 'static,
+{
+    instance.set_storage_readonly(false);
+    call_raw(
+        instance,
+        "migrate",
+        &[env, msg, migrate_info],
+        read_limits::RESULT_MIGRATE,
+    )
+    .or_else(|err| {
+        if matches!(err, VmError::FunctionArityMismatch { .. }) {
+            call_raw(
+                instance,
+                "migrate",
+                &[env, msg],
+                read_limits::RESULT_MIGRATE,
+            )
+        } else {
+            Err(err)
+        }
+    })
+}
```

Comment:
Yeah, exactly – this is super nice. All users of cosmwasm-vm can benefit from the feature now and we have strong typing